actions/generate-schedule/scheduler/scheduler.go

```diff
@@ -33,70 +34,64 @@ func NewScheduler(userSource users.Source, shiftDurationDays int) (*Scheduler, e
 	}, nil
 }
 
-// Schedule creates a new Schedule from start (inclusive) to stop (exclusive). Will return a 1-entry schedule if
-// stop is before start.
+// Schedule creates a new Schedule that includes whole shifts of `Scheduler.shiftDuration` from start (inclusive) to
+// stop (inclusive).  Will return an error if stop is before start, or either start are stop are zero values.
 func (s *Scheduler) Schedule(start, stop time.Time) (*schedule.Schedule, error) {
-	sched := &schedule.Schedule{
-		Shifts: []*schedule.Shift{
-			{
-				StartDate: start.Format(DateFormat),
-				User:      s.userSource.NextUser(),
-			},
-		},
+	if start.IsZero() || stop.IsZero() {
+		return nil, fmt.Errorf("neither start (%v) nor stop (%v) can be zero values", start, stop)
+	}
+
+	if stop.Before(start) {
+		return nil, fmt.Errorf("start cannot be before stop")
 	}
-	if err := s.ExtendSchedule(sched, stop); err != nil {
-		return nil, err
+
+	sched := &schedule.Schedule{}
+	s.extendSchedule(sched, start, stop)
+
+	if err := sched.Validate(); err != nil {
+		return nil, fmt.Errorf("error validating new schedule: %v", err)
```

Comment:
I feel like `extendSchedule` should do the validation itself?

Reply:
Fair point, moved.